### ISSUE TITLE
GNOME 40 Update Issue

### ISSUE THREAD

**Issue by LittleFire000000000** [home-sweet-gnome/dash-to-panel#1331]
Hello!!  I'm a big fan of Dash to Panel.  But Gnome 40's Extensions app isn't.  It keeps trying to update me to the version that worked on Fedora 33, but I now run 34.  Every time Extensions is opened and then I later logout or reboot, when I log back in, the F33 version is loaded.  The app thinks your working F34 version is older than the official version.  It calls this an update.  I have to reinstall your version every time I open Extensions.  I'm not a Javascript or Gnome developer; I can't give you a fix.  But can you make the version number higher than the now broken F33 version so that Extensions would stop giving my F34 system the now broken F33 version?

**Comment by jderose9** [comment]
@philippun1's latest changes have been merged to master. If you install using these instructions you should get the GNOME 40 support and not have the Extensions updater write over top of it:

https://github.com/home-sweet-gnome/dash-to-panel/wiki/Installation#installation-from-github

**Comment by LittleFire000000000** [comment]
> Hi @LittleFire000000000. We will have to wait until the original developer upgrades it to work with gnome 40.
> Meanwhile try this
> 
> `git clone -b "update-to-gnome40" https://github.com/philippun1/dash-to-panel.git dash-to-panel`
> 
> in the Makefile & metadata.json: replace the UUID with something different but same In Both
> 
> And finally
> 
> `cd dash-to-panel && make && make install`

This works excellently.  I had to change the username in the Makefile.  Otherwise, it works consistently.  Thanks, Daring.

**Comment by philippun1** [comment]
@LittleFire000000000 I just installed the latest master version with the instructions from the wiki page and it worked flawlessly. Are you sure you have pulled the latest master?

**Comment by reagentoo** [comment]
> All of your setup tricks don't work for me:

I think I understand the reason. Gentoo Linux has the following settings:
```
$ cat /etc/eselect/gnome-shell-extensions/eselect-gnome-shell-extensions.gschema.override 
# Generated and used by eselect gnome-shell-extensions; do not edit manually
[org.gnome.shell]
enabled-extensions=[]
```
This was done in order to be able to selectively disable installed extensions on the system. 
I didn't have to change anything here before. But now I have to do the following:
```
# eselect gnome-shell-extensions enable dash-to-panel@jderose9.github.com

$ cat /etc/eselect/gnome-shell-extensions/eselect-gnome-shell-extensions.gschema.override 
# Generated and used by eselect gnome-shell-extensions; do not edit manually
[org.gnome.shell]
enabled-extensions=['dash-to-panel@jderose9.github.com']
```
@philippun1 any ideas?

**Comment by LittleFire000000000** [comment]
> @LittleFire000000000 I just installed the latest master version with the instructions from the wiki page and it worked flawlessly. Are you sure you have pulled the latest master?

@philippun1 Your version works, but Extensions still updates it to the broken version.  I retested it to confirm.

**Comment by philippun1** [comment]
@LittleFire000000000 As said before: please use the latest home-sweet-gnome/dash-to-panel master branch. My branch has been merged and the update issue is only fixed in upstream!

**Comment by jderose9** [comment]
I think that ideally we would have the following issues resolved before releasing the GNOME 3.40 release to extensions.gnome.org. If anyone is able to contribute time to looking into any of these it would be a big help:

#1326 - Probably the most critical
#1338
#1327 - Not really important but probably solved with 1338.
#1333 - if the workaround PR is "good enough" then maybe this is solved

There are a few features not working such as the custom panel color, gradient and anything else set with inline CSS but they are less used features with workaround (such as setting in your theme) so I think we can just remove those for the initial release.
